src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -212,20 +197,29 @@ export async function filterLogEventsFromUriComponents(
 ): Promise<CloudWatchLogsResponse> {
     const client = new DefaultCloudWatchLogsClient(logGroupInfo.regionName)
 
-    const response = await client.filterLogEvents({
+    const timeout = new Timeout(10000)
+    showMessageWithCancel(`Loading log data from group ${logGroupInfo.groupName}`, timeout)
+    const responsePromise = client.filterLogEvents({
         logGroupName: logGroupInfo.groupName,
         filterPattern: parameters.filterPattern,
         nextToken,
         limit: parameters.limit,
     })
 
+    const response = await waitTimeout(responsePromise, timeout, { allowUndefined: false })
```

Comment:
You can throw if `response` is `void`
The types for `waitTimeout` are wrong (my bad):
```suggestion
    const response = await waitTimeout(responsePromise, timeout, { allowUndefined: false })
    if (!response) {
        throw new Error('`filterLogEvents` did not return anything')
    }
```

Reply:
Wait sorry, what about the types are wrong for `waitTimeout`?

Reply:
`waitTimeout` is incorrectly saying that it can return `void` when it never can.